src/Libraries/CoreNodes/Data.cs

```diff
@@ -51,16 +70,114 @@ private static object ToNative(JToken token)
             }
         }
 
+        /// <summary>
+        /// Parse implementation for converting JObject types to specific Dynamo objects (ie Geometry, Color, Images, etc) 
+        /// </summary>
+        /// <param name="jObject"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        private static object DynamoJObjectToNative(JObject jObject)
+        {
+            if (jObject.ContainsKey("$typeid"))
+            {
+                var typeid = jObject["$typeid"].ToString();
+
+                switch (typeid)
+                {
+                    //autodesk.math to abstract ProtoGeometry types
+                    case "autodesk.math:vector3d-1.0.0":
+                        return Vector.FromJson(jObject.ToString());
+
+                    case "autodesk.math:matrix44d-1.0.0":
+                        return CoordinateSystem.FromJson(jObject.ToString());
+
+                    //autodesk.geometry to abstract ProtoGeometry types
+                    case "autodesk.geometry:boundingbox3d-1.0.0":
+                        return BoundingBox.FromJson(jObject.ToString());
+
+                    case "dynamo.geometry:mesh-1.0.0":
+                        return Mesh.FromJson(jObject.ToString());
+
+                    //types supported by Goemetry.FromJson
+                    case "autodesk.math:point3d-1.0.0":
+                    case "dynamo.geometry:sab-1.0.0":
+                    case "dynamo.geometry:tsm-1.0.0":
+                    case "dynamo.geometry:rectangle-1.0.0":
+                    case "dynamo.geometry:cuboid-1.0.0":
+                    case "dynamo.geometry:solid-1.0.0":
+                    case string geoId when geoId.Contains("autodesk.geometry"):
+                        return Geometry.FromJson(jObject.ToString());
+
+                    //Dynamo types
+                    case "dynamo.graphics:color-1.0.0":
```

Comment:
for all of these dynamo schemas, where do I find them?

Reply:
do you think it makes sense to provide these schemas in the repo?

Reply:
I don't think we need the schemas here yet.  But also I don't think we should break.

Reply:
do you the tests cover all schemas?

Reply:
Yes